### PR TITLE
Deploy SA with 3 separate app registrations, Fixes #686

### LINKED ISSUE #686: New App registration for the portal app
**Is your feature request related to a problem? Please describe.**
Add new App Registration for customer portal application

**Describe the solution you'd like**
Add new App Registration for customer portal application

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by neelavarshad**
Fixes #686 

Creates unique App Registrations for both landing page and admin portal
Associates appropriate App registrations and URI redirects
Added both app registrations to web app config
Added additional Client ID to supporting interface

### CODE REVIEW

**Review by bkhabazan — APPROVED** [review]

File: src/CustomerSite/appsettings.Development.json.rename
[the remark this answers is not in the document]
Comment (santhoshb-msft) [reply src/CustomerSite/appsettings.Development.json.rename:1]: working on this internally

File: deployment/Deploy.ps1 [hunk 1]
```diff
@@ -188,7 +189,17 @@ if($LogoURLico) {
 #region Create AAD App Registrations
 
 #Record the current ADApps to reduce deployment instructions at the end
-$ISADMTApplicationIDProvided = $ADMTApplicationID
+$ISADMTApplicationIDProvided = ($ADMTApplicationIDAdmin && $ADMTApplicationIDPortal)
+
+if($ISADMTApplicationIDProvided -eq $null){
+	Write-Host "🔑 Multi-Tenant App Registrations provided."
```
Comment (santhoshb-msft) [inline deployment/Deploy.ps1:195]: Maybe we can call it , "Login App Registration.."

File: deployment/Deploy.ps1 [hunk 2]
```diff
@@ -211,6 +222,84 @@ if (!($ADApplicationID)) {
...
+    try {
+	
+		$appCreateRequestBodyJson = @"
+{
+	"displayName" : "$WebAppNamePrefix-AdminPortalAppReg",
+	"api": 
+	{
+		"requestedAccessTokenVersion" : 2
+	},
+	"signInAudience" : "AzureADandPersonalMicrosoftAccount",
```
Comment (santhoshb-msft) [inline deployment/Deploy.ps1:237]: This should be single Tenant

File: src/AdminSite/Startup.cs [hunk 3]
```diff
@@ -96,7 +97,7 @@ public void ConfigureServices(IServiceCollection services)
             .AddOpenIdConnect(options =>
             {
                 options.Authority = $"{config.AdAuthenticationEndPoint}/common/v2.0";
```
Comment (santhoshb-msft) [inline src/AdminSite/Startup.cs:99]: this should change to /tenantid/ not common

**Review by santhoshb-msft — COMMENTED** [review]
Lets review when you get a chance please

File: deployment/Deploy.ps1 (hunk 2 above)
Comment (neelavarshad) [reply deployment/Deploy.ps1:237]: This is a good way to restrict users further. That said, sometimes the individual responsible for actions on the admin portal may not even be part of the tenant hosting the app registrations. In those cases, partner will need to invite the user to the tenant before being able to manage subscriptions.

File: src/AdminSite/Startup.cs (hunk 3 above)
Comment (neelavarshad) [reply src/AdminSite/Startup.cs:99]: Will make this change too if you no longer want multitenant access to the admin portal.

File: deployment/Deploy.ps1
Comment (santhoshb-msft) [inline deployment/Deploy.ps1:1]: --sign-in-audience AzureADMYOrg

**Pull request merged**